packages/hint-create-element-svg/src/hint.ts

```diff
@@ -0,0 +1,96 @@
+         * a 'createElement' property. If yes, check if that call expression's
+         * first argument is an SVG element. In that case, report an error as
+         * the correct identifier property for creating SVG elements is
+         * 'createElementNS'
+         */
+        linter.defineRule('svg-create', {
+            create(eslintContext) {
+                return {
+                    CallExpression(node) {
+                        const exp = node as ESTree.CallExpression;
+                        const args = exp.arguments;
+                        const callee = exp.callee as ESTree.MemberExpression;
+                        const property = callee !== undefined && callee !== null ? callee.property as ESTree.Identifier : null;
+
+                        if (property !== null && property.name === 'createElement' && args.length >= 1 && args[0].type === 'Literal') {
+                            const arg = args[0] as ESTree.Literal;
+
+                            if (arg.value !== undefined && arg.value !== null && svgElements.has(arg.value.toString().toLowerCase())) {
+                                eslintContext.report({
+                                    messageId: 'avoidElement',
+                                    node: property
+                                });
+                            }
+                        }
+                    }
+                };
+            },
+            meta: { messages: { avoidElement: 'Avoid using createElement to create SVG elements; use createElementNS instead' } }
+        });
+
+        let scriptElement: IAsyncHTMLElement;
+
+        const validateScript = async (scriptData: ScriptParse) => {
+
+            debug(`Validating hint create-element-svg`);
+
+            const sourceCode = scriptData.sourceCode;
+            const results = linter.verify(sourceCode, { rules: { 'svg-create': 'error' } });
+
+            for (const result of results) {
+                let position: ProblemLocation | null = scriptElement.getLocation() || null;
+                let line = 0;
+
+                /*
+                 * Eslint returns location data starting at <script>
+                 * This offsets the line number so that it is relative
+                 * to the full source code and not just the <script> tag
+                 */
+                if (scriptElement) {
+                    position = await findProblemLocation(scriptElement, { column: 0, line: 0 }, result.source === null ? '' : result.source);
+                    line = position !== null ? position.line : 0;
+                }
+                await context.report(scriptData.resource, result.message, { location: { column: result.column, line: line + result.line }});
+            }
+        };
+
+        const enterScript = (elementFound: ElementFound) => {
+            scriptElement = elementFound.element;
+        };
+
+        context.on('element::script', enterScript);
```

Comment:
This is more of a nit and @molant may disagree, but I'd favor registering anonymous functions directly for both `element::script` and `parse::end::javascript` since we only register once and never deregister. This also allows TypeScript to correctly infer parameter types so you have to import even fewer things. For example:
```suggestion
        context.on('element::script', (elementFound) => {
            scriptElement = elementFound.element;
        });
```

Reply:
Not going to fight you on this one 😛